### DISCUSSION TITLE
OSP Molecules catalogue

### DISCUSSION THREAD

**Discussion by jgonlop3** [Open-Systems-Pharmacology/Forum#820] [Feature Requests & Ideas]
Hi, I´m new in OSP and I think that a good idea for new users like me is to have a catalogue of molecules with preconfigured properties. I think that PK-Sim can be very usefull for clinical uses with this information.

**Comment by msevestre** [comment]
Great idea

**Comment by jgonlop3** [comment]
I don´t have sufficient programming background kwnoledge but I´m very interested in this topic. How can I help? Ideas to lead and planning work?

**Comment by msevestre** [comment]
@jgonlop3 
The major task first of all is to compile this list of ever growing preconfigured molecules. 


For each molecule, following parameters are required:
- Name,
- Lipophilicity
- MolWeight
- Solubility at Ref pH
- Ref pH
- PKAs

Then ideally, some processes describing how the drugs interacts with different enzymes, transporters etc..

The actual format could be an 
* a json file : [here](https://github.com/Open-Systems-Pharmacology/InstallationValidator/blob/develop/data/Inputs/BatchFiles/Human_MixedInhibition.json#L20) is an example of how a compound is read by pk-sim to perform batch simulations).
```
{
  "Name": "inhibitor",
  "Lipophilicity": 2,
  "FractionUnbound": 0.6,
  "MolWeight": 3.0E-7,
  "SolubilityAtRefpH": 1.0E-7,
  "RefpH": 9,
  "PkaTypes": [{
      "Type": "Acid",
      "Value": 8
  }],
   "PartialProcesses": [{
      "MoleculeName": "CYP3A4",
      "InternalName": "MixedInhibition",
      "DataSource": "Lab",
      "ParameterValues": {"Ki_c": 1,"Ki_u": 1}
  }],
  "CalculationMethods": ["Cellular partition coefficient method - Rodgers and Rowland"]
}
```
* a template database file: It is possible to export a compound to a user template database
* ...

Having a json file with all the compound properties would offer many advantages. One of which would be that this file would be on github and anyone could participate and add/modify the catalogue of molecules

So far, no programming knowledge required. Only  a Text editor  :)

In PK-Sim,  a simple change would have to be implemented to allow for
* Export of one or more compound to json (so that it can be added to the catalog)
* Import a compound from the molecule catalog

**Comment by jgonlop3** [comment]
Perfect! I think it´s easy to do. I start today! I make an LibreOffice Calc file with this properties. I think Calc is more easy for non-programmers than json. I agree with your comment, the perfect situation will be the PK-sim option to import/export properties of molecules.

**Comment by msevestre** [comment]
@Yuri05 @TheiBa Do you have any suggestions ?

**Comment by TheiBa** [comment]
What about the idea to import this information, e.g. from 
https://www.drugbank.ca/

I think that would offer accepted, referencable start values. Could involve 2 
steps: 1. importing/extracting the information from above into a suitable 
format that is 2. readable by PK-Sim?

**Comment by TheiBa** [comment]
and further qualified models will be developed and posted as discussed in #11

**Comment by msevestre** [comment]
I fear that importing compound data without validating them may lead to
quite a lot of unusable substances.
Do you think we can use values from the drug bank in PK-Sim as is? It gets
even more interesting when we think about processes.

**Comment by TheiBa** [comment]
<https://github.com/notifications/beacon/AYQo2fcKCrP5b6brigYBewoeCkFcYtdsks5r62lxgaJpZM4NeICW.gif> 
No, I am pretty sure we cannot expect the resulting models to have a high 
quality, but they can serve as an easy start that then needs refinement and 
qualification/validation for intended use. But I guess that is more the topic 
of #11?!

So, I see two topics: easy access to reasonable phys-chem values to start 
modelling (here), and then further developing quality models fit for purpose 
(#11).

**Comment by jgonlop3** [comment]
Yes, I am agree with @msevestre. No quality = no usefull. But it´s a good starting point like @TheiBa told. There is a lot of databases information about molecular properties but we need to use the properties that match the best results with OSP. I think about starting with some of this references values and then validate the whole model in PK-Sim. The pack of properties for any molecule must be coordinate with adequate package of anatomy+physiological values...etc. In summary, we need to stablish an initial molecular values that works with the reference PBPK model implemmented in PK-Sim.

**Comment by msevestre** [comment]
@jgonlop3 
>I make an LibreOffice Calc file with this properties

Be aware that the number of properties is not set. That's why an excel/calc table won't work.
For example, in the json text I pasted above, you can see that there is also a `PartialProcess` defined for `CYP3A4` of type `MixedInhibition`.  
A compound can have many different processes defined and as you can imagine, this will be very hard to model in a table.

**Comment by jgonlop3** [comment]
Yes, I know the limitations of a table. I try to do a database (sqlite or libreoffice base are good options for this purpose). First of all I need to design the tables and relations. I think on spreadsheet like Calc previously because of its simplicity. But I think you are right and database is better option. I will make a database.

**Comment by msevestre** [comment]
@jgonlop3 the problem with a database is that it is very hard to extend if you are non programmer. 
Why not try the json file option? This is a simple text file that can be edited in almost any decent text editor.
The beauty in this approach is that anyone could submit a patch on github and it would be fairly easy to see what has change between two versions (text file)

At the moment however, you can start with a simple table as you suggested which would have to be converted to another format once somewhat ready. Thoughts?

**Comment by jgonlop3** [comment]
@msevestre I don´t like json format because it seems programming language code! difficult to read for me: not too intuitive... but you´re right again! is simple and this is the key. I will do it in json.

**Comment by msevestre** [comment]
Also another cool advantage is that pksim could load this file (in start or
on demand) and use it as a remote compound database (json files are ideal
for that...very lightweight)

**Comment by msevestre** [comment]
See https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/168 for upcoming implementation in PK-Sim that will allow users to create a json block automatically

**Comment by jgonlop3** [comment]
Thank you @msevestre  I´ll read it

**Comment by jgonlop3** [comment]
I have a doubt: I am writing a json for vancomycin. I take the properties from Chemicalize.com. I have a lot of usefull information but I don´t know where to write it inside json frame. @msevestre can you help me? I try to simplify  and take only the most important properties but in the json example you had sent I don´t know where to place this information. Example: How can I write solubility at differents pH? I think it is a very  easy for you... ;)

**Comment by msevestre** [comment]
@jgonlop3 
What PKSim is supporting now is fairly limited

For example you can only enter one solubility for a given pH.  Those properties are called `SolubilityAtRefpH` and `RefpH` respectively. We may have to extend the json format to support all capability of PK-Sim in the future.  For example 

{
  "Name": "inhibitor",
  "Lipophilicity": 2,
  "FractionUnbound": 0.6,
  "MolWeight": 3.0E-7,
  **"SolubilityAtRefpH"**: 1.0E-7,
  **"RefpH"**: 9,
...
}

**Comment by jgonlop3** [comment]
I think about this idea. Is it planned that PK-Sim can export data of
molecules in json format? Because we can use PK-Sim to do a template that
users can load with different data. Secondly, do a warehouse of molecules
and then we do planning to validate them. Each set of data for a molecule
(we can do many sets for the same molecule) can include information about
it validation status.

2017-05-19 17:29 GMT+02:00 Michael Sevestre <notifications@github.com>:

> @jgonlop3 <https://github.com/jgonlop3>
> What PKSim is supporting now is fairly limited
>
> For example you can only enter one solubility for a given pH. Those
> properties are called SolubilityAtRefpH and RefpH respectively. We may
> have to extend the json format to support all capability of PK-Sim in the
> future. For example
>
> {
> "Name": "inhibitor",
> "Lipophilicity": 2,
> "FractionUnbound": 0.6,
> "MolWeight": 3.0E-7,
> *"SolubilityAtRefpH"*: 1.0E-7,
> *"RefpH"*: 9,
> ...
> }
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Open-Systems-Pharmacology/Forum/issues/10#issuecomment-302734693>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AEQnqUFeQytYIw5-SpjDj8hznKvdU8gZks5r7bVdgaJpZM4NeICW>
> .
>



-- 
*Jaime González López*

**Comment by jgonlop3** [comment]
We can use PK-sim as frontend. We only need export capabilities.

2017-05-19 17:37 GMT+02:00 Jaime Gonzalez <jgonlop3@gmail.com>:

> I think about this idea. Is it planned that PK-Sim can export data of
> molecules in json format? Because we can use PK-Sim to do a template that
> users can load with different data. Secondly, do a warehouse of molecules
> and then we do planning to validate them. Each set of data for a molecule
> (we can do many sets for the same molecule) can include information about
> it validation status.
>
> 2017-05-19 17:29 GMT+02:00 Michael Sevestre <notifications@github.com>:
>
>> @jgonlop3 <https://github.com/jgonlop3>
>> What PKSim is supporting now is fairly limited
>>
>> For example you can only enter one solubility for a given pH. Those
>> properties are called SolubilityAtRefpH and RefpH respectively. We may
>> have to extend the json format to support all capability of PK-Sim in the
>> future. For example
>>
>> {
>> "Name": "inhibitor",
>> "Lipophilicity": 2,
>> "FractionUnbound": 0.6,
>> "MolWeight": 3.0E-7,
>> *"SolubilityAtRefpH"*: 1.0E-7,
>> *"RefpH"*: 9,
>> ...
>> }
>>
>> —
>> You are receiving this because you were mentioned.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Open-Systems-Pharmacology/Forum/issues/10#issuecomment-302734693>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AEQnqUFeQytYIw5-SpjDj8hznKvdU8gZks5r7bVdgaJpZM4NeICW>
>> .
>>
>
>
>
> --
> *Jaime González López*
>



-- 
*Jaime González López*

**Comment by jgonlop3** [comment]
Regarding solubility data: How can I write 2 solubilities at 2 different pHs? Can you give me an example?

**Comment by msevestre** [comment]
@jgonlop3 
>Is it planned that PK-Sim can export data of molecules in json format?

yes: https://github.com/Open-Systems-Pharmacology/PK-Sim/issues/168
This will be released in 7.2.0. As soon as the feature is implemented however, you'll be able to try it out with the pre release

>Regarding solubility data: How can I write 2 solubilities at 2 different pHs? Can you give me an example?

This is not supported by PK-Sim. Instead, using RefpH and Ref Solubility and solubility gain per charge, PK-Sim predicts the Sol  = f(pH)

Here is an extract of Chapter 15 describing how PK-Sim handles solubility
>The solubility can be specified together with the type of measurement or the medium used (first column, Experiment). The corresponding unit can be chosen from the drop-down menu in the second column (Solubility at Ref-pH). For charged compounds, the pH value at which the solubility of the compound was measured should be given in the third column (Ref-pH). In the fourth column, the Solubility gain per charge can be modified, which defines the factor by which the solubility increases with each ionization step. In order to calculate the charge of the molecule, the fraction of each microspecies is calculated according to the Henderson-Hasselbalch equation for a given pH. This is done across the entire pH-range such that the fractions are used to calculate the probability with which a molecule is in a certain ionization state. Based on this information, the pH-dependent solubility of molecules with one or more ionizable groups is calculated. By clicking on Show Graph, the pH-dependent solubility across the whole pH range calculated based on the experimental solubility at the defined pH is shown. For neutral compounds the input fields Ref-pH and Solubility gain per charge and the graph are irrelevant.

**Comment by jgonlop3** [comment]
Thanks @msevestre

**Comment by jgonlop3** [comment]
In PK-Sim I can select differents units but in the json file you sent me as example this is not specified. Where can I registered the units (example mg/L) of the solubility at Ref-pH inside json files? 

 "SolubilityAtRefpH": 0.177827,   <- Which units?????
  "RefpH": 7.4,

**Comment by msevestre** [comment]
This is a very good point. The json specification does not support unit at the moment. That means that for now, all values should be defined in base unit for a specific dimension.
PK-Sim and MoBi are using the following dimension and unit definition

https://github.com/Open-Systems-Pharmacology/OSPSuite.Dimensions/blob/master/OSPSuite.Dimensions.xml


For example, Concentration (molar) as a base unit of umol/l
https://github.com/Open-Systems-Pharmacology/OSPSuite.Dimensions/blob/master/OSPSuite.Dimensions.xml#L165

When the export is implemented in PK-Sim,. values will be exported in the correct unit by default

Does it make sense?

**Comment by jgonlop3** [comment]
Understand. Is possible to add dime coment in json file with the units? Only like explanation. Ir nota is possible, dont worry. I understand It.

**Comment by PavelBal** [comment]
Why json? I am not a fan of XML, but wouldn't it be more consistent to use XML for compound definitions? OSPS uses XML for exporting model blocks and whole simulations (MatLab/R), so no new structure should be thought of. Furthermore, schemas could help in validating the molecules catalogue.

The advantage of json of being more lighweight should not play any role as the data should be used locally.

**Comment by Yuri05** [comment]
Defining a schema would be a good idea in any case.
JSON format also supports schema definition and validation can be performed even online.
http://www.jsonschemavalidator.net/

**Comment by msevestre** [comment]
* json is the format that we use for batch simulations (see links above). We already support this format out of the box
* json is much more readable than xml
* any new API now favors json over xml
* and it will be loaded right from github.com (albeit with a fallback option local). We will use this file a standard api endpoint to retrieve the compound list online

That being said, the format we have now, that I copied as an example above, will probably need to be extended. More particularly, I think we need to add

* unit for parameter value (this will serve for documentation purpose)
* ValueDescription (text describing where the value is coming from. Absolute must so that this compound value can be trusted)
* description of compound itself
* ...probably much more
*  Alternative values (a way to define more than one lipophilicity for example with different source)
* fu: For which species ?

@jgonlop3 While json will remain, the actual data structure will change now that we go from an intern use to external use only. It might be wise to wait a bit until the structure is finalize

**Comment by CcVn** [comment]
Dear all,

The proposal of an "incremental" approach made by TheiBa sounds reasonable. Drugbank is a good source to start with : I used it quite often as a starting point when building a new PBPK model from scratch on reference compounds, and the quality of information seems sufficient for a first version of a model. Then the compound file may be gradually completed/optimized to use other validated in vitro or ex vivo data found elsewhere. Up to a real "validation" against clinical data (first PK in healthy people, then in disease population etc.). This approach may require to have a dedicated and standardized field within the compound file indicating the degree/scale of quality/validation of the file, starting with the lowest grade (pure in silico data) to the highest grade (file validated with various clinical datasets). This way people could know what degree of confidence they can put into their simulations.

Best regards,

**Comment by StephanSchaller** [comment]
@jgonlop3 , do you have any updates on this topic? Did you manage to create a database?

**Reply by IulianIV** [reply]
Hello!
                                                                                                                                                                                                                                                    
Is this still being worked on, or is there interest in a community-maintained compound catalog?
                                                                                                                                                                                                                                                    
I've been exploring `PKSimDB.sqlite` and the existing model JSONs (e.g. the Digoxin model's Compound section) along with the roposed JSON format from this discussion. I understand the parameter schema and data sourcing. I already used data curated from **DrugBank**, **PubChem**, and **ChEMBL** to reconstruct the Digoxin compound parameters and managed to closely match the actual model data.
                                                                                                                                                                                                                                                    
I'd be happy to take this on if it's not actively being worked on.

**Comment by martinobelvederi** [comment]
i'm very interested in this idea, especially on psychotropic drug modelling - though I'm not a pharmacologist, but I will try to involve some colleagues who are. Anyway I wonder if in the meantime this idea led to any other repository or similar initiatives

**Reply by Yuri05** [reply]
Hi - great question, and the good news is that a lot has happened since this thread was first opened in 2017. <br>The "preconfigured molecules catalogue" idea has effectively been realized through **two complementary, openly maintained resources**. You no longer need to hand-build a JSON catalogue to get going - both can be used directly in PK-Sim today.

## 1. The OSP PBPK Model Library - fully validated, ready-to-use models

The [**OSP PBPK Model Library**](https://www.open-systems-pharmacology.org/OSP-PBPK-Model-Library/) is the curated, version-controlled catalogue of **validated** compound/PBPK models. It is exactly the "validated warehouse" that was discussed here - each entry has gone through a documented evaluation against clinical PK data.

For every compound you get:

* an **HTML evaluation report** (model assumptions, parameters, data sources, goodness-of-fit),
* a downloadable **PDF report**, and
* ready-to-open **PK-Sim project file(s)** (`.pksim5`).

The library is released in sync with the software (currently **v12.3**) and is organized by therapeutic area, e.g.:

* **Antibiotics / antivirals / antifungals:** Amikacin, Clarithromycin, Erythromycin, Rifampicin, Vancomycin, Atazanavir, Efavirenz, Raltegravir, Omeprazole, …
* **Anesthetics & analgesics:** Alfentanil, Sufentanil, …
* **Sedatives, anxiolytics & CNS-relevant compounds:** Alprazolam, Midazolam, Triazolam, Carbamazepine, S-Mephenytoin, Fluvoxamine, Moclobemide, Tizanidine, …
* **Cardiovascular & others:** Cimetidine, Digoxin, Dapagliflozin, Felodipine, Verapamil, Sildenafil, Montelukast, Propofol, Mexiletine, …
* **Antibodies & biologics:** 7E3, BAY794620, CDA1, dAb2, Inulin, MEDI524, MEDI524YTE, Tefibazumab, ...

Each model carries a defined "degree of validation" via its evaluation report - which is precisely the quality-grading concern raised earlier in this thread.

> **For psychotropic-drug modelling specifically:** the library and templates already cover several CNS-active / psychotropic-relevant compounds (e.g. Alprazolam, Midazolam, Triazolam, Carbamazepine, Fluvoxamine, Moclobemide, Desipramine, Atomoxetine). These are excellent, peer-reviewed starting points for building or extending models in that space.

<img width="1133" height="457" alt="image" src="https://github.com/user-attachments/assets/5555041d-a64c-4f63-b629-2e92ce039a7d" />

## 2. Compound templates inside PK-Sim - including extra published models

PK-Sim itself now ships with a **"Load Compound from template"** function (Building Block Explorer → Compounds → *Load From Template* → **Template Source**). This is the practical, no-coding "import a compound from the catalog" capability that was requested in this very thread.

<img width="1118" height="986" alt="image" src="https://github.com/user-attachments/assets/ad179aa5-1602-4d96-beba-33f57fab8a43" />

A key point - and this is what the screenshot above shows - is that the template list is **broader than the Model Library**. The dialog has an **"OSP Library Release"** column:

* **Yes** → the compound is part of the officially released, validated OSP PBPK Model Library (e.g. Alfentanil, Alprazolam, Amikacin, Atazanavir, Carbamazepine, Clarithromycin, Digoxin, Efavirenz, …).
* **No** → the compound is an **additional published model** (e.g. released as a **supplement to a journal publication**) that is **not (yet) part of the OSP-PBPK-Model-Library**, but is still made available as a loadable template - for example **Atomoxetine, Coproporphyrin I, Desipramine, Furosemide, Ketoconazole**.

So between the two, you have access to *more* compounds than the Library alone. Each template links back to its source model repository (the `Url` column, e.g. `https://github.com/Open-Systems-Pharmacology/<Compound>-Model`). You can also tick **"Only show OSP library released templates"** if you want to restrict yourself to the fully validated set (which is the **default** PK-Sim setting).

## 3. Where to look for even more (and how to contribute)

* **OSP documentation** - for the compound workflow and how phys-chem/ADME parameters are defined and used: https://docs.open-systems-pharmacology.org/
* [**OSP-based publications list**](https://github.com/Open-Systems-Pharmacology/OSP-based-publications-and-content/issues?q=is%3Aopen+is%3Aissue+label%3AJournal) - a continuously updated index of peer-reviewed publications and models built with OSP; many of these correspond to the "Library Release = No" templates and are great leads for additional compounds (including more CNS/psychotropic examples).

The original JSON/batch format mentioned earlier in this thread still exists (it's how PK-Sim reads compounds for batch simulations), so the community-contribution path is also alive: a new compound model is typically developed, validated, published, and added as its own `*-Model` repository, from where it becomes loadable as a template and - once formally evaluated - promoted into the Model Library.

**Reply by martinobelvederi** [reply]
@Yuri05 sorry for the late reply - this is really fantastic! can't wait to try it out .. thanks so much!
